ansible/modules/hashivault/hashivault_identity_group.py

```diff
@@ -0,0 +1,264 @@
+            group_details = client.secrets.identity.read_group(group_id=group_id)
+        except Exception as e:
+            return {'failed': True, 'msg': str(e)}
+        return hashivault_identity_group_update(group_details['data'], client, group_id, group_name, group_type,
+                                                 group_metadata, group_policies, group_member_group_ids,
+                                                 group_member_entity_ids, mount_point)
+    elif group_name is not None:
+        try:
+            group_details = client.secrets.identity.read_group_by_name(name=group_name)
+        except Exception:
+            response = client.secrets.identity.create_or_update_group_by_name(
+                name=group_name,
+                group_type=group_type,
+                metadata=group_metadata,
+                policies=group_policies,
+                member_group_ids=group_member_group_ids,
+                member_entity_ids=group_member_entity_ids,
+                mount_point=mount_point
+            )
+            return {'changed': True, 'data': response.json()['data']}
+        return hashivault_identity_group_update(group_details['data'], client, group_name=group_name,
+                                                 group_id=group_details['data']['id'],
+                                                 group_type=group_type,
+                                                 group_metadata=group_metadata,
+                                                 group_policies=group_policies,
+                                                 group_member_group_ids=group_member_group_ids,
+                                                 group_member_entity_ids=group_member_entity_ids,
+                                                 mount_point=mount_point)
+    return {'failed': True, 'msg': "Either name or id must be provided"}
+
+
+def hashivault_identity_group_delete(params):
+    client = hashivault_auth_client(params)
+    group_id = params.get('id')
+    group_name = params.get('name')
+
+    if group_id is not None:
+        try:
+            client.secrets.identity.read_group(group_id=group_id)
+        except Exception:
+            return {'changed': False}
+        client.secrets.identity.delete_group(group_id=group_id)
+        return {'changed': True}
+    elif group_name is not None:
+        try:
+            client.secrets.identity.read_group_by_name(name=group_name)
+        except Exception:
+            return {'changed': False}
+        client.secrets.identity.delete_group_by_name(name=group_name)
+        return {'changed': True}
+    return {'failed': True, 'msg': "Either name or id must be provided"}
+
+
+@hashiwrapper
+def hashivault_identity_group(params): 
+    state = params.get('state')
+    if state == 'present':
+        return hashivault_identity_group_create_or_update(params)
+    elif state == 'absent':
+        return hashivault_identity_group_delete(params)
+    else:
```

Comment:
this is a unnecessary check since your ansible play limits the input suggestion already to 'present' or 'absent' 

`argspec['state'] = dict(required=False, choices=['present', 'absent'], default='present')`

Reply:
got it - removed.